sdk/core/azure-core/src/main/java/com/azure/core/util/DateTimeRfc1123.java

```diff
@@ -46,6 +52,75 @@ public OffsetDateTime getDateTime() {
         return this.dateTime;
     }
 
+    /**
+     * Parses the RFC1123 format datetime string into OffsetDateTime.
+     *
+     * @param date The datetime string in RFC1123 format
+     * @return The underlying OffsetDateTime.
+     */
+    private OffsetDateTime parse(final String date) {
+        return OffsetDateTime.of(
```

Comment:
Have an early out for if the `date` string is less than 25 characters

Reply:
what about let the offsetDateTime.of() to check it.